pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -982,7 +982,20 @@ func (r *MachineSyncReconciler) fetchCAPIInfraResources(ctx context.Context, cap
 //nolint:funlen,gocognit,cyclop
 func (r *MachineSyncReconciler) reconcileMAPItoCAPIMachineDeletion(ctx context.Context, mapiMachine *machinev1beta1.Machine, capiMachine *capiv1beta1.Machine, infraMachine client.Object) (bool, error) {
 	if mapiMachine.DeletionTimestamp.IsZero() {
-		return false, nil
+		if capiMachine == nil || capiMachine.DeletionTimestamp.IsZero() {
+			// Neither MAPI authoritative machine nor its CAPI non-authoritative machine mirror
+			// are being deleted, nothing to reconcile for deletion.
+			return false, nil
+		}
+
+		// The MAPI authoritative machine is not being deleted, but the CAPI non-authoritative one is.
+		// Issue a deletion also to the MAPI authoritative machine.
+		if err := r.Client.Delete(ctx, mapiMachine); err != nil {
+			return false, fmt.Errorf("failed to delete Machine API machine: %w", err)
+		}
+
+		// Return true to force a requeue, to allow the deletion propagation.
+		return true, nil
```

Comment:
The delete on the mapi machine will add an event to the queue, I don't think that requeueing actually does anything

Reply:
Yes, the `true` there doesn't add a new event on the queue, But does just return and wait for an event triggered by the Delete. This is a snipped of the logic is doing in the caller of this func:

https://github.com/openshift/cluster-capi-operator/blob/ac5aa33f14d786ee00a8ed1647a7e99ccdb9f017/pkg/controllers/machinesync/machine_sync_controller.go#L411-L415